libsolidity/parsing/Parser.cpp

```diff
@@ -2076,16 +2078,39 @@ optional<string> Parser::findLicenseString(std::vector<ASTPointer<ASTNode>> cons
 	for (auto const& [start, end]: sequencesToSearch)
 	{
 		smatch match;
-		if (regex_search(start, end, match, licenseRegex))
+		auto words_begin =
+			std::sregex_iterator(start, end, licenseRegex);
+		auto words_end = std::sregex_iterator();
+
+		for (std::sregex_iterator i = words_begin; i != words_end; ++i)
 		{
-			string license{boost::trim_copy(string(match[1]))};
-			if (!license.empty())
+			std::smatch match = *i;
+			if (!match.empty())
+			{
+				string license{string(match[1])};
+				// Remove trailing comment terminator if present
+				if (boost::algorithm::ends_with(license, "*/"))
+					license = license.substr(0, license.size() - 2);
+
+				license = boost::trim_copy(license);
```

Comment:
I don't understand the `?:` part of your regex suggestion

Reply:
Also, doesn't the `([^\n]+) eat all the characters anyway so what follows will be ignored?

Reply:
At least with this, the regex doesn't properly work so far :)

Reply:
`(?:)` is a non-capturing group. Same as `()` but the content does not appear on the list of matches.

Sorry, looks like I did not escape the regex properly (I should have tried it on something like https://regex101.com first). Should have been:
```c++
"SPDX-License-Identifier:\\s*([^\n]+)\\s*(?:\*\/)?"
```

> Also, doesn't the `([^\n]+)` eat all the characters anyway so what follows will be ignored?

It does capture the `*/` but you get it as a part of your group and then you have to cut it out manually. My regex would match it and then discard it.

Reply:
I saw the escaping problems and did that myself, but the regex still won't work. The `([^\n]+)` will simply capture everything until the end of the line...

Reply:
See also https://javascript.info/regexp-greedy-and-lazy

Reply:
Ah, you're right. It would work if `*/` or whitespace was not optional. With the current structure the `([^\n]+)` part grabs them instead.

Ok then. I guess it might be possible to come up with a more complex regex to handle it but at this point cutting it out manually starts looking like the better approach :)

Reply:
That was my thinking ^^ however, by turning my `([^\n]+)` into a `([^\n])+?` it actually works (switching it from greedy to lazy)

Reply:
But won't `([^\n])+?`, being non-greedy, grab only the first character of the license name and stop?

I mean, it should work with regex match  where the whole string has to be matched but with regex search I'd expect it to stop after it gets the smallest possible match.